### DISCUSSION TITLE
New Wifi credentials are not saved after reboot, ESP8266 always take the same old credentials

### DISCUSSION THREAD

**Discussion by undee123** [khoih-prog/ESPAsync_WiFiManager#91] [General]
Hello to everybody,

I've tried to get ESPAsync_WiFiManager to work by suing only the minimal requirements and coding:
```
#include <ESPAsync_WiFiManager.h> //https://github.com/khoih-prog/ESPAsync_WiFiManager
AsyncWebServer webServer(80);
DNSServer dnsServer;
 
 
void setup()
{
    // put your setup code here, to run once:
    Serial.begin(115200);
    while (!Serial)
        ;
    delay(200);
    Serial.print("\nStarting Async_AutoConnect_ESP8266_minimal on " + String(ARDUINO_BOARD));
    Serial.println(ESP_ASYNC_WIFIMANAGER_VERSION);
    ESPAsync_WiFiManager ESPAsync_wifiManager(&webServer, &dnsServer, "My-ESP8266");
    Serial.println("Connect to previously saved AP...");
    Serial.println( WiFi.SSID() );
    Serial.println( WiFi.psk() );
    ESPAsync_wifiManager.autoConnect("My-ESP8266");
 
    if (WiFi.status() == WL_CONNECTED)
    {
        Serial.print(F("Connected. Local IP: "));
        Serial.println(WiFi.localIP());
    }
    else
    {
        Serial.println(ESPAsync_wifiManager.getStatus(WiFi.status()));
        Serial.println("Can't connect! Enter WiFi config mode...");
        Serial.println("Restart...");
        ESP.reset();
    }
 
 
}
 
void loop()
{
}
```
Connecting to the ESP's access point for WiFi configuration works perfectly, and the ESP connects to the new WiFi, but after a reboot the connection does not work anymore and I have to configure it again.
I've discovered that the credentials the ESP uses after reboot are some old ones from a WiFi that doesn't exist anymore:
```
Serial.println( WiFi.SSID() );
Serial.println( WiFi.psk() );
```
Why does it not save the new credentials? Where the error?

Thank you for any help!!!

**Comment by khoih-prog** [comment]
> What can you expect from the minimal version of examples?
> 
> It's not designed to save the Credentials, just to show how to shorten the code, certainly with loosing all the necessary features.

Check [**Doesn't save network credentials** #58](https://github.com/khoih-prog/ESPAsync_WiFiManager/issues/58) and **don't use any `minimal` example**

I already deleted all `minimal` examples in latest releases v1.12.2

**Comment by undee123** [comment]
Hi Khoih,

thank you for your replay (and thank you of course for this library!!!), I'll try to find out which minimal code lines I need to get the credentials stored.

Greetings from Italy!

**Comment by undee123** [comment]
I'm sorry, I really can't find the error in my code.
Tried with "normal" WifiManager with this minimal code:
```
#include <WiFiManager.h> // https://github.com/tzapu/WiFiManager

void setup() {
    WiFi.mode(WIFI_STA);
    Serial.begin(115200);
    WiFiManager wm;

    bool res;
    res = wm.autoConnect("AutoConnectAP","password"); // password protected ap

    if(!res) {
        Serial.println("Failed to connect");
    } 
    else {
        Serial.println("connected...yeey :)");
    }
}

void loop() {
}
```
And here the SSID and Passphrase are stored. I really don't understand why, hope someone can help me please :-|
